api/python/quilt3/data_transfer.py

```diff
@@ -295,20 +302,50 @@ def _download_file(ctx, src_bucket, src_key, src_version, dest_path):
 
     dest_file.parent.mkdir(parents=True, exist_ok=True)
 
+    with dest_file.open('wb', buffering=0) as f:
+        fileno = f.fileno()
+        is_regular_file = stat.S_ISREG(os.stat(fileno).st_mode)
```

Comment:
`dest_file.is_file()`

Reply:
I presume that calling `os.stat()` with a file descriptor is faster than with a path, since the latter requires path resolving.

Reply:
Ok sure